implementations/rust/ockam/ockam_api/src/session/mod.rs

```diff
@@ -118,6 +121,7 @@ impl Medic {
                                 session.set_status(Status::Down);
                                 log::info!(%key, "replacing session");
                                 self.replacements.spawn(async move { (key, f.await) });
+                                let _ = sleep(self.retry_delay).await;
```

Comment:
If we sleep here, the whole `Medic` instance would be frozen for 15 seconds, since the async context is shared for every session.
If we move the sleep inside the spawned async context on line 123 that would block only that specific session.